repos/system_upgrade/common/actors/detectmissingnewlineingrubcfg/actor.py

```diff
@@ -0,0 +1,37 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.libraries.actor.detectmissingnewlineingrubcfg import detect_config_error
+from leapp.models import GrubConfigError
+from leapp.reporting import create_report, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class DetectMissingNewLineInGrubCfg(Actor):
+    """
+    Check grub configuration for missing new line at its end.
+    """
+
+    name = 'detect_missing_new_line_in_grub_cfg'
+    consumes = ()
+    produces = (Report, GrubConfigError)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        config = '/etc/default/grub'
+        error_detected = detect_config_error(config)
+        if error_detected:
+            create_report([
+                reporting.Title('Detected a missing new line at the end of grub configuration file.'),
+                reporting.Summary(
+                    'The missing new line in /etc/default/grub causes booting issues when appending '
+                    'new entries to this file during the upgrade. '
+                    'Error is automatically fixed by add_upgrade_boot_entry actor.'
+                ),
+                reporting.Severity(reporting.Severity.LOW),
+                reporting.Tags([reporting.Tags.BOOT]),
+                reporting.RelatedResource('file', config)
+            ])
+
+        config_error = GrubConfigError(error_detected=error_detected,
+                                       error_description='missing LF')
+        self.produce(config_error)
```

Comment:
The design of the model does not make much sense. If there is no problem with configuration, don't produce anything.
```suggestion
        if error_detected:
            config_error = GrubConfigError(error_detected=error_detected, error_description='missing LF')
            self.produce(config_error)
```

Reply:
You are right, the boolean is redundant and should be removed.